File: P_2/Napakalaki_Ruby_P2/lib/treasure.rb
#encoding: utf-8
# MADE BY: JOSE MIGUEL HDEZ GARCIA & JAIRO ABRIL MOYA

class Treasure
    
    #Consultores basicos
    attr_reader :name
    attr_reader :bonus
    attr_reader :type
    
    # Constructor
    def initialize(name = "", bonus = 0, type = Array.new)
        @name = name
        @bonus = bonus
        @type = type
    end
    
    # Metodos
    def to_s
        "Nombre: #{@name}
         Bonus: #{@bonus.to_s}
         Tipo: #{@type.to_s}"
    end
end

File: P_2/Examen_P2/Napakalaki_Java_P2/src/napakalaki/Treasure.java
// MADE BY: JOSE MIGUEL HDEZ GARCIA & JAIRO ABRIL MOYA

package napakalaki;

/**
 *
 * @author ixjosemi
 */

public class Treasure {

    private String name;
    private int bonus;
    private TreasureKind type;


    public Treasure (String name, int bonus, TreasureKind type){
        this.name = name;
        this.bonus = bonus;
        this.type = type;
    }
    
    public String getName(){
        return name;
    }
    
    public int getBonus(){
        return bonus;
    }
    
    public TreasureKind getType(){
        return type;
    }
    
}
